panoramapublic/test/src/org/labkey/test/components/panoramapublic/PanoramaPublicSearchWebPart.java

```diff
@@ -0,0 +1,154 @@
+import org.labkey.test.components.html.Input;
+import org.labkey.test.util.DataRegionTable;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class PanoramaPublicSearchWebPart extends BodyWebPart<PanoramaPublicSearchWebPart.ElementCache>
+{
+    private static String title = "Panorama Public Search";
+
+    public PanoramaPublicSearchWebPart(WebDriver driver, String title)
+    {
+        super(driver, title);
+    }
+
+    @Override
+    protected PanoramaPublicSearchWebPart.ElementCache newElementCache()
+    {
+        return new PanoramaPublicSearchWebPart.ElementCache();
+    }
+
+    public String getTitle()
+    {
+        return elementCache().title.get();
+    }
+
+    public PanoramaPublicSearchWebPart setTitle(String title)
+    {
+        elementCache().title.set(title);
+        return this;
+    }
+
+    public String getAuthors()
+    {
+        return elementCache().authors.get();
+    }
+
+    public PanoramaPublicSearchWebPart setAuthors(String value)
+    {
+        elementCache().authors.set(value);
+        return this;
+    }
+
+    public String getOrganism()
+    {
+        return elementCache().organism.get();
+    }
+
+    public PanoramaPublicSearchWebPart setOrganism(String value)
+    {
+        elementCache().organism.set(value);
+        return this;
+    }
+
+    public String getInstrument()
+    {
+        return elementCache().instrument.get();
+    }
+
+    public PanoramaPublicSearchWebPart setInstrument(String value)
+    {
+        elementCache().instrument.set(value);
+        return this;
+    }
+
+    public String getProtein()
+    {
+        return elementCache().protein.get();
+    }
+
+    public PanoramaPublicSearchWebPart setProtein(String value)
+    {
+        elementCache().protein.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setProteinExactMatch(boolean value)
+    {
+        elementCache().proteinExactMatch.set(value);
+        return this;
+    }
+
+    public String getPeptide()
+    {
+        return elementCache().peptide.get();
+    }
+
+    public PanoramaPublicSearchWebPart setPeptide(String value)
+    {
+        elementCache().peptide.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart setPeptideExactMatch(boolean value)
+    {
+        elementCache().peptideExactMatch.set(value);
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoExperimentSearch()
+    {
+        getWrapper().click(Locator.tagWithText("span", "Experiment Search"));
+        getWrapper().waitForElementToBeVisible(Locator.name("Authors"));
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoProteinSearch()
+    {
+        elementCache().proteinTab.click();
+        return this;
+    }
+
+    public PanoramaPublicSearchWebPart gotoPeptideSearch()
+    {
+        elementCache().peptideTab.click();
+        return this;
+    }
```

Comment:
Is there something to wait for when switching to protein or peptide search? Similar to `gotoExperimentSearch` waiting for the "Authors" to be visible.

Reply:
I had added initially, but it was obsolete so removed it.